cpp/src/arrow/compute/exec/aggregate_node.cc

```diff
@@ -282,12 +396,19 @@ class ScalarAggregateNode : public ExecNode, public TracedNode {
     auto scope = TraceInputReceived(batch);
     DCHECK_EQ(input, inputs_[0]);
 
-    auto thread_index = plan_->query_context()->GetThreadIndex();
-
-    ARROW_RETURN_NOT_OK(DoConsume(ExecSpan(batch), thread_index));
+    auto handler = [this](const ExecBatch& full_batch, const GroupingSegment& segment) {
+      if (!segment.extends && segment.offset == 0) RETURN_NOT_OK(OutputResult());
+      auto exec_batch = full_batch.Slice(segment.offset, segment.length);
+      auto batch = ExecSpan(exec_batch);
+      RETURN_NOT_OK(DoConsume(batch, plan_->query_context()->GetThreadIndex()));
```

Comment:
What is the difference between ExecSpan and ExecBatch here and why do we remove the ExecSpan?

Reply:
Oh wait u didn't, you just named the ExecSpan to "batch" which confused me

Reply:
This seems unnecessary changes, before the code was

```
    auto thread_index = plan_->query_context()->GetThreadIndex();
    ARROW_RETURN_NOT_OK(DoConsume(ExecSpan(batch), thread_index));
```

```
      auto batch = ExecSpan(exec_batch);
      RETURN_NOT_OK(DoConsume(batch, plan_->query_context()->GetThreadIndex()));
```

Seems exactly the same thing?

Reply:
Before, `thread_index` was not within a lambda. I'll fix this to be similar to the original code.